content/docs/15.how-to-guides/build-custom-plugin.md

```diff
@@ -0,0 +1,425 @@
+
+#### Create a new plugin
+
+Here are the steps:
+
+1. Go on the [plugin-template](https://github.com/kestra-io/plugin-template) repository.
+2. Click on *Use this template*.
+3. Choose the GitHub account you want to link and the repository name for the new plugin.
+4. Clone the new repository: `git clone git@github.com:{{user}}/{{name}}.git`.
+5. Open the cloned directory in IntelliJ IDEA.
+6. Enable [annotations processors](https://www.jetbrains.com/help/idea/annotation-processors-support.html).
+7. If you are using an IntelliJ IDEA < 2020.03, install the [lombok plugins](https://plugins.jetbrains.com/plugin/6317-lombok) (if not, it's included by default).
+
+Once you completed the steps above, you should see a similar directory structure:
+
+![Structure](/docs/plugin-developer-guide/plugins-architecture.png)
+
+As you can see, there is one generated plugin: the `Example` class representing the `Example` plugin (a task).
+
+A project typically hosts multiple plugins. We call a project a group of plugins, and you can have multiple sub-groups inside a project by splitting plugins into different packages. Each package that has a plugin class is a sub-group of plugins.
+
+## Gradle Configuration
+
+We use [Gradle](https://gradle.org/) as a build tool.
+
+### Mandatory configuration
+
+The first thing we need to configure is the plugin name and the class package.
+
+1. Change in `settings.gradle` the `rootProject.name = 'plugin-template'` with the plugin name `rootProject.name = 'plugin-pokemon'`.
+2. Change the class package: by default, the template provides a package `io.kestra.plugin.templates`, just rename the `templates` folder in `src/main/java` & `src/test/java` to `pokemon`. And change the first line in `Example.java`, `ExampleRunnerTest.java` and `ExampleTest.java` to `package io.kestra.plugin.pokemon;`.
+3. In `build.gradle`: 
+   a. replace `description 'Plugin template for Kestra'` to the package name `description 'Plugin pokemon for Kestra'`.
+   b. In the `dependencies` section, add a dependency which we will be using in our plugin task:
+    `implementation group: 'com.googlecode.json-simple', name: 'json-simple', version: '1.1.1'` 
+   c. Change the `jar` section to the following:
+
+    ```groovy
+    jar {
+        manifest {
+            attributes(
+                    "X-Kestra-Name": project.name,
+                    "X-Kestra-Title": "Pokemon",
+                    "X-Kestra-Group": project.group + ".pokemon",
+                    "X-Kestra-Description": project.description,
+                    "X-Kestra-Version": project.version
+            )
+        }
+    }
+    ```
+
+## Develop Fetch Task
+
+### Create Pokemon class
+
+In `src/main/java/io/kestra/plugin/pokemon`, we will create a new class `Pokemon.java`. This will be used to map the JSON output of the pokemon API to the Java class. We can put in only those fields that we are interested in, and ignore the rest.
```

Comment:
```suggestion
In `src/main/java/io/kestra/plugin/pokemon`, we will create a new class `Pokemon.java`. This will be used to map the JSON output of the pokemon API to the Java class. We only need to add the fields that we are interested in, and ignore the rest.
```